my-sms/spec/controllers/students_controller_spec.rb

```diff
@@ -23,119 +25,114 @@
-      get :new, {}, valid_session
-      expect(response).to be_success
+    context '#new' do
+      it 'returns a success response' do
+        get :new, {}
+        expect(response).to be_success
+      end
     end
-  end
 
-  describe "GET #edit" do
-    it "returns a success response" do
-      student = Student.create! valid_attributes
-      get :edit, {:id => student.to_param}, valid_session
-      expect(response).to be_success
+    context '#edit' do
+      it 'returns a success response' do
+        get :edit, { id: subject.to_param }
+        expect(response).to be_success
+      end
     end
   end
 
-  describe "POST #create" do
-    context "with valid params" do
-      it "creates a new Student" do
-        expect {
-          post :create, {:student => valid_attributes}, valid_session
-        }.to change(Student, :count).by(1)
+  describe 'POST #create' do
+    let(:student_args) { {} }
+    subject do
+      student = FactoryBot.attributes_for(:student, **student_args)
+      post :create, { student: student }
+    end
+
+    context 'with valid params' do
+      it 'creates a new Student' do
+        expect { subject }.to change(Student, :count).by(1)
       end
 
-      it "redirects to the created student" do
-        post :create, {:student => valid_attributes}, valid_session
-        expect(response).to redirect_to(Student.last)
+      it 'redirects to the created student' do
+        # subject
+        expect(subject).to redirect_to(Student.last)
       end
     end
 
-    context "with invalid params" do
+    context 'with invalid params' do
+      let(:student_args) { { first_name: nil } }
+
       it "returns a success response (i.e. to display the 'new' template)" do
-        post :create, {:student => invalid_attributes}, valid_session
-        expect(response).to be_success
+        expect(subject).to be_success
       end
     end
   end
 
-  describe "PUT #update" do
-    context "with valid params" do
-      let(:new_attributes) {
-        skip("Add a hash of attributes valid for your model")
-      }
-
-      it "updates the requested student" do
-        student = Student.create! valid_attributes
-        put :update, {:id => student.to_param, :student => new_attributes}, valid_session
-        student.reload
-        skip("Add assertions for updated state")
+  describe 'PUT #update' do
+    let(:new_student_args) { {} }
+    subject do
+      student = FactoryBot.create(:student)
+      new_attributes = FactoryBot.attributes_for(:student, **new_student_args)
+      put :update, { id: student.to_param, student: new_attributes }
+    end
+
+    context 'with valid params' do
+      it 'updates the requested student' do
+        subject do
+          student.reload
+          attrs = %i[first_name last_name gender]
+          student_attrs = student.attributes.fetch_values(*attrs.map(&:to_s))
+          new_attrs = new_attributes.fetch_values(*attrs)
+          expect(student_attrs).to eq new_attrs
+        end
```

Comment:
This doesn't need to be wrapped in the subject here, but you will need to call the subject 

```suggestion
        subject
        student.reload
        attrs = %i[first_name last_name gender]
        student_attrs = student.attributes.fetch_values(*attrs.map(&:to_s))
        new_attrs = new_attributes.fetch_values(*attrs)
        expect(student_attrs).to eq new_attrs

```